charts/fluent-bit/templates/pdb.yaml

```diff
@@ -0,0 +1,20 @@
+{{- if and  ( eq  .Values.kind "Deployment" )  .Values.podDisruptionBudget.enabled  }}
+apiVersion: policy/v1beta1
```

Comment:
```suggestion
apiVersion: {{ include "fluent-bit.pdb.apiVersion" . }}
```

I think you need to define a function in _helpers.tpl_ to set the correct `apiVersion` as the old version will show warnings from K8s v1.21.

```tpl
{{/* Get PodDisruptionBudget API Version */}}
{{- define "fluent-bit.pdb.apiVersion" -}}
  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">= 1.21-0" .Capabilities.KubeVersion.Version) -}}
      {{- print "policy/v1" -}}
  {{- else -}}
    {{- print "policy/v1beta1" -}}
  {{- end -}}
{{- end -}}
```

Reply:
How come you decided to inline the logic rather than implementing the template I suggested?

Reply:
done inline without helper

Reply:
I'd prefer to see this in a helper method, it lowers the cognitive load to understand the manifest.

Reply:
as you wish